dashboard/app/controllers/programming_classes_controller.rb

```diff
@@ -58,6 +59,15 @@ def show
     @programming_environment_categories = @programming_class.programming_environment.categories_for_navigation
   end
 
+  def show_by_keys
```

Comment:
Could we add a comment with the URL that gets routed here (like we do in some other parts of the code)?  It'd be nice if we could find this method by searching for the beginning of the URL.

Same for ProgrammingExpression.  Also ok to do this in a separate PR if it feels unrelated.

Reply:
I'll add for these actions in this PR as they're new but it would be good for me to add it for all in a separate PR